File: Scripts/results/generated_code_results/RN/Social Networking/SocialMessagingApp/ChatScreen.js
import React, { useState } from 'react';
import { View, Text, Image, TextInput, TouchableOpacity, FlatList, StyleSheet } from 'react-native';
import { useNavigation } from '@react-navigation/native';
const ChatScreen = () => {
const navigation = useNavigation();
const [message, setMessage] = useState('');
const messages = [
{ id: '1', text: 'Hi! Your last shot was really good!', time: '9:30', sender: 'user' },
{ id: '2', type: 'audio', duration: '1.35', time: '9:32', sender: 'contact' },
{ id: '3', text: 'What tools do you use?', time: '9:34', sender: 'user' },
{ id: '4', text: 'Figma, for prototype i use Principle', time: '9:35', sender: 'contact' },
{ id: '5', text: 'Cool! Your designs inspire me a lot bro. Keep posting!', time: '9:36', sender: 'user' },
{ id: '6', text: 'Thank u so much Antonio! Glad to hear that :)', time: '9:37', sender: 'contact' },
{ id: '7', text: "You're welcome!", time: '9:39', sender: 'user' },
];
const renderMessage = ({ item }) => (
<View style={[styles.messageBubble, item.sender === 'user' ? styles.userMessage : styles.contactMessage]}>
{item.type === 'audio' ? (
<View style={styles.audioMessage}>
<View style={styles.waveform} />
<Text style={styles.audioDuration}>{item.duration}</Text>
</View>
) : (
<Text style={styles.messageText}>{item.text}</Text>
)}
<Text style={styles.messageTime}>{item.time}</Text>
</View>
);
return (
<View style={styles.container}>
<View style={styles.header}>
<TouchableOpacity onPress={() => navigation.goBack()}>
<Image source={require('../assets/snack-icon.png')} style={styles.backIcon} />
</TouchableOpacity>
<Image source={require('../assets/snack-icon.png')} style={styles.profilePicture} />
<View>
<Text style={styles.userName}>Antonio Banderas</Text>
<Text style={styles.userStatus}>Online</Text>
</View>
<TouchableOpacity onPress={() => navigation.navigate('Profile')}>
<Image source={require('../assets/snack-icon.png')} style={styles.optionsIcon} />
</TouchableOpacity>
</View>
  <FlatList
    data={messages}
    renderItem={renderMessage}
    keyExtractor={(item) => item.id}
    contentContainerStyle={styles.messageList}
  />

  <View style={styles.inputContainer}>
    <TouchableOpacity>
      <Image source={require('../assets/snack-icon.png')} style={styles.inputIcon} />
    </TouchableOpacity>
    <TextInput
      style={styles.input}
      placeholder="Type a message"
      value={message}
      onChangeText={setMessage}
    />
    <TouchableOpacity>
      <Image source={require('../assets/snack-icon.png')} style={styles.inputIcon} />
    </TouchableOpacity>
    <TouchableOpacity>
      <Image source={require('../assets/snack-icon.png')} style={styles.inputIcon} />
    </TouchableOpacity>
  </View>
</View>
);
};
const styles = StyleSheet.create({
container: {
flex: 1,
backgroundColor: '#fff',
},
header: {
flexDirection: 'row',
alignItems: 'center',
padding: 10,
borderBottomWidth: 1,
borderBottomColor: '#e0e0e0',
},
backIcon: {
width: 24,
height: 24,
marginRight: 10,
},
profilePicture: {
width: 40,
height: 40,
borderRadius: 20,
marginRight: 10,
},
userName: {
fontSize: 18,
fontWeight: 'bold',
},
userStatus: {
fontSize: 14,
color: 'green',
},
optionsIcon: {
width: 24,
height: 24,
marginLeft: 'auto',
},
messageList: {
paddingVertical: 10,
},
messageBubble: {
maxWidth: '80%',
padding: 10,
borderRadius: 20,
marginVertical: 5,
},
userMessage: {
alignSelf: 'flex-end',
backgroundColor: '#dcf8c6',
},
contactMessage: {
alignSelf: 'flex-start',
backgroundColor: '#fff',
borderWidth: 1,
borderColor: '#e0e0e0',
},
messageText: {
fontSize: 16,
},
messageTime: {
fontSize: 12,
color: '#888',
alignSelf: 'flex-end',
marginTop: 5,
},
audioMessage: {
flexDirection: 'row',
alignItems: 'center',
},
waveform: {
width: 100,
height: 30,
backgroundColor: '#e0e0e0',
borderRadius: 15,
},
audioDuration: {
marginLeft: 10,
fontSize: 14,
color: '#888',
},
inputContainer: {
flexDirection: 'row',
alignItems: 'center',
padding: 10,
borderTopWidth: 1,
borderTopColor: '#e0e0e0',
},
input: {
flex: 1,
height: 40,
borderWidth: 1,
borderColor: '#e0e0e0',
borderRadius: 20,
paddingHorizontal: 10,
marginHorizontal: 10,
},
inputIcon: {
width: 24,
height: 24,
},
});
export default ChatScreen;

File: Scripts/results/generated_code_results/Flutter/Social Networking/SocialMessagingApp/ChatScreen.dart
import 'package:flutter/material.dart';

class ChatScreen extends StatefulWidget {
@override
_ChatScreenState createState() => _ChatScreenState();
}

class _ChatScreenState extends State<ChatScreen> {
final List<Map<String, dynamic>> messages = [
{'text': 'Hi! Your last shot was really good!', 'time': '9:30', 'sender': 'user'},
{'type': 'audio', 'duration': '1.35', 'time': '9:32', 'sender': 'contact'},
{'text': 'What tools do you use?', 'time': '9:34', 'sender': 'user'},
{'text': 'Figma, for prototype i use Principle', 'time': '9:35', 'sender': 'contact'},
{'text': 'Cool! Your designs inspire me a lot bro. Keep posting!', 'time': '9:36', 'sender': 'user'},
{'text': 'Thank u so much Antonio! Glad to hear that :)', 'time': '9:37', 'sender': 'contact'},
{'text': "You're welcome!", 'time': '9:39', 'sender': 'user'},
];

@override
Widget build(BuildContext context) {
return Scaffold(
appBar: AppBar(
leading: IconButton(
icon: Icon(Icons.arrow_back),
onPressed: () => Navigator.pop(context),
),
title: GestureDetector(
onTap: () => Navigator.pushNamed(context, 'Profile'),
child: Row(
children: [
CircleAvatar(
backgroundImage: AssetImage('assets/images/template.png'),
radius: 20,
),
SizedBox(width: 8),
Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
Text('Antonio Banderas'),
Text('Online', style: TextStyle(fontSize: 12, color: Colors.green)),
],
),
],
),
),
actions: [
IconButton(icon: Icon(Icons.more_vert), onPressed: () {}),
],
),
body: Column(
children: [
Expanded(
child: ListView.builder(
itemCount: messages.length,
itemBuilder: (context, index) {
final message = messages[index];
return MessageBubble(
text: message['text'] ?? '',
time: message['time'] ?? '',
isUser: message['sender'] == 'user',
isAudio: message['type'] == 'audio',
audioDuration: message['duration'] ?? '',
);
},
),
),
Container(
padding: EdgeInsets.symmetric(horizontal: 8),
child: Row(
children: [
IconButton(icon: Icon(Icons.emoji_emotions), onPressed: () {}),
Expanded(
child: TextField(
decoration: InputDecoration(
hintText: 'Type a message',
border: InputBorder.none,
),
),
),
IconButton(icon: Icon(Icons.attach_file), onPressed: () {}),
IconButton(icon: Icon(Icons.send), onPressed: () {}),
],
),
),
],
),
);
}
}

class MessageBubble extends StatelessWidget {
final String text;
final String time;
final bool isUser;
final bool isAudio;
final String audioDuration;

MessageBubble({
required this.text,
required this.time,
required this.isUser,
this.isAudio = false,
this.audioDuration = '',
});

@override
Widget build(BuildContext context) {
return Align(
alignment: isUser ? Alignment.centerRight : Alignment.centerLeft,
child: Container(
margin: EdgeInsets.symmetric(vertical: 4, horizontal: 8),
padding: EdgeInsets.all(12),
decoration: BoxDecoration(
color: isUser ? Colors.blue : Colors.grey[300],
borderRadius: BorderRadius.circular(20),
),
child: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
if (isAudio)
Row(
mainAxisSize: MainAxisSize.min,
children: [
Icon(Icons.play_arrow, color: isUser ? Colors.white : Colors.black),
SizedBox(width: 8),
Container(
width: 100,
height: 4,
color: isUser ? Colors.white : Colors.black,
),
SizedBox(width: 8),
Text(audioDuration, style: TextStyle(color: isUser ? Colors.white : Colors.black)),
],
)
else
Text(text, style: TextStyle(color: isUser ? Colors.white : Colors.black)),
SizedBox(height: 4),
Text(time, style: TextStyle(color: isUser ? Colors.white70 : Colors.black54, fontSize: 12)),
],
),
),
);
}
}